### DISCUSSION TITLE
Using variable fonts

### DISCUSSION THREAD

**Discussion by gaborpapp** [Chlumsky/msdf-atlas-gen#82] [Q&A] [answered]
I see that variable fonts are supported with the  `-varfont` parameter, but I cannot figure out how to specify which typeface to use from the font file. Any help would be appreciated.

**Comment by gaborpapp** [comment]
I tried setting something like this: `-varfont font.otf?wght=900&wdth=200`, I had to escape `?` and `&`, because I'm not on Windows.
Whatever I did I got the same default result. So I ended up converting the font instance to a static font with `fonttools`.

**Reply by Chlumsky** [reply]
Can you provide the font so I can test it?

**Reply by gaborpapp** [reply]
Sure. Is it ok if I send a link by email?

**Reply by Chlumsky** [reply]
No problem.

**Reply by Chlumsky** [reply, marked as answer]
The variable axes of this font are named
- `Width` (50 to 200),
- `Weight` (200 to 900),
- and `ital` (0 to 1).

So you can use for example `-varfont "Obviously.otf?Width=100&Weight=500&ital=0.5"`

**Reply by gaborpapp** [reply]
Thanks a lot! This indeed works.
May I ask how you figured out the axes? Both https://wakamaifondue.com/ and `ttx` lists `wdht`, `wght` and `ital`.


    
    <Axis>
      <AxisTag>wdth</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>50.0</MinValue>
      <DefaultValue>50.0</DefaultValue>
      <MaxValue>200.0</MaxValue>
      <AxisNameID>263</AxisNameID>
    </Axis>

    
    <Axis>
      <AxisTag>wght</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>200.0</MinValue>
      <DefaultValue>200.0</DefaultValue>
      <MaxValue>900.0</MaxValue>
      <AxisNameID>264</AxisNameID>
    </Axis>

    
    <Axis>
      <AxisTag>ital</AxisTag>
      <Flags>0x0</Flags>
      <MinValue>0.0</MinValue>
      <DefaultValue>0.0</DefaultValue>
      <MaxValue>1.0</MaxValue>
      <AxisNameID>265</AxisNameID>
    </Axis>

**Reply by Chlumsky** [reply]
I got it directly from FreeType's `FT_Get_MM_Var` listed as `FT_Var_Axis::name`. It's though possible that this is not the best source of the axis names for the library so if anyone has a better suggestion, let me know.

**Reply by gaborpapp** [reply]
The problem is that they don't seem to match with other sources, but since we need to use the axis names for Freetype anyway, it would be a nice feature to add a command line parameter to list axis names.

**Comment by Chlumsky** [comment]
The 4 letter tags (like `wght` instead of `Weight`) are now recognized and all axis names, ranges and defaults can now be listed using

```
msdfgen -varfont fontname.otf printvars
```

or

```
msdf-atlas-gen -varfont fontname.otf -printvaraxes
```

**Comment by gaborpapp** [comment]
Awesome. Thank you!
